ig/ig.go

```diff
@@ -0,0 +1,172 @@
+package ig
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+)
+
+type IG struct {
+	path  string
+	image string
+	v1    int
+	v2    int
+	v3    int
+}
+
+type option func(*IG)
+
+func Path(path string) option {
+	return func(ig *IG) {
+		ig.path = path
+	}
+}
+
+func Image(image string) option {
+	return func(ig *IG) {
+		ig.image = image
+	}
+}
```

Comment:
In the functional options pattern the options are usually called with a `With` prefix -> `WithPath`, `WithImage`, etc.